src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs

```diff
@@ -311,19 +311,20 @@ public async Task Verify_Indicator_Display_On_Selection()
 
 			await UnitTestUIContentHelperEx.SetContentAndWait(SUT);
 
-		
+
 
 			for (int i = 0; i < NumItems; i++)
 			{
 				SUT.SelectedIndex = i;
 				await UnitTestsUIContentHelper.WaitForIdle();
 
-				var selectedItem = SUT.ContainerFromItem(SUT.SelectedItem) as TabBarItem;
+				var container = SUT.ContainerFromItem(SUT.SelectedItem);
+				var selectedItem = SUT.GetInnerContainer(container);
```

Comment:
No, it returns a `ContentPresenter` from `ContainerFromItem` and `GetInnerContainer` will return the `TabBarItem`. I pushed a test to ensure we wouldn't not have nested TBI, but instead a CP with a TBI.

https://github.com/unoplatform/uno.toolkit.ui/blob/c44ac4e7f850044657344d2c697b7c2aad86dafb/src/Uno.Toolkit.RuntimeTests/Tests/TabBarTests.cs#L430-L463

Reply:
i would add a comment/xcomment on GetInnerContainer in the TabBar.cs
and, everywhere we are using it, add a comment, to check the above comment
it would avoid the confusion here, which will greatly benefit _future us_